src/parser/mod.rs

```diff
@@ -8192,11 +8192,25 @@ impl<'a> Parser<'a> {
     /// Parses a column definition within a view.
     fn parse_view_column(&mut self) -> Result<ViewColumnDef, ParserError> {
         let name = self.parse_identifier(false)?;
-        let options = if dialect_of!(self is BigQueryDialect | GenericDialect)
+        let mut options = vec![];
+        if dialect_of!(self is BigQueryDialect | GenericDialect)
             && self.parse_keyword(Keyword::OPTIONS)
         {
             self.prev_token();
-            Some(self.parse_options(Keyword::OPTIONS)?)
+            if let Some(option) = self.parse_optional_column_option()? {
+                options.push(option);
+            }
+        };
+        if dialect_of!(self is SnowflakeDialect | GenericDialect)
```

Comment:
```suggestion
        } else if dialect_of!(self is SnowflakeDialect | GenericDialect)
```
nit: maybe an else if to show that both conditions do the same for different dialects

Reply:
Updated with 
```
if (dialect_of!(self is BigQueryDialect | GenericDialect)
            && self.parse_keyword(Keyword::OPTIONS))
            || (dialect_of!(self is SnowflakeDialect | GenericDialect)
                && self.parse_keyword(Keyword::COMMENT))
        {
```
by `clippy` message.